api/app/controllers/spree/api/base_controller.rb

```diff
@@ -188,6 +189,17 @@ def paginate(resource)
       def default_per_page
         Kaminari.config.default_per_page
       end
+
+      def invalid_transition(error)
+        logger.error("invalid_transition #{error.event} from #{error.from} for #{error.object.class.name}. Error: #{error.inspect}")
+        if error.object.is_a? Spree::Order
```

Comment:
I don't get this, why we should print a deprecation message if an order is passed? I think the deprecation should only be present in the order `.jbuilder` error partial, that we will not use anymore in Solidus, and will be shown if someone is using that file in their own extension/store.